server/src/frameworks/Foundry/FoundryProject.ts

```diff
@@ -0,0 +1,286 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { analyze } from "@nomicfoundation/solidity-analyzer";
+import fs from "fs";
+import { CompilerInput } from "hardhat/types";
+import _ from "lodash";
+import path from "path";
+import semver from "semver";
+import { DidChangeWatchedFilesParams } from "vscode-languageserver-protocol";
+import { analyzeSolFile } from "../../parser/analyzer/analyzeSolFile";
+import { OpenDocuments, ServerState } from "../../types";
+import { toUnixStyle } from "../../utils";
+import directoryContains from "../../utils/directoryContains";
+import { getOrInitialiseSolFileEntry } from "../../utils/getOrInitialiseSolFileEntry";
+import { runCmd } from "../../utils/operatingSystem";
+import CompilationDetails from "../base/CompilationDetails";
+import Project from "../base/Project";
+import { Remapping } from "./Remapping";
+
+interface SolFileDetails {
+  path: string;
+  pragmas: string[];
+}
+
+export default class FoundryProject extends Project {
+  public priority = 1;
+  public sourcesPath!: string;
+  public testsPath!: string;
+  public remappings: Remapping[] = [];
+  public initializeError?: string;
+  public configSolcVersion?: string;
+
+  constructor(
+    serverState: ServerState,
+    basePath: string,
+    public configPath: string
+  ) {
+    super(serverState, basePath);
+  }
+
+  public id(): string {
+    return this.configPath;
+  }
+
+  public frameworkName(): string {
+    return "Foundry";
+  }
+
+  public async initialize(): Promise<void> {
+    try {
+      const config = JSON.parse(
+        await runCmd("~/.foundry/bin/forge config --json", this.basePath)
+      );
+      this.sourcesPath = path.join(this.basePath, config.src);
+      this.testsPath = path.join(this.basePath, config.test);
+      this.configSolcVersion = config.solc || undefined; // may come as null otherwise
+
+      const rawRemappings = await runCmd(
+        "~/.foundry/bin/forge remappings",
```

Comment:
Very recently I changed `forge` to `~/.foundry/bin/forge` to have it work on mac. It seems to have worked because the CI is able to run the foundry integration test now. This is on the server side only. After this comment I noticed we are also using forge on the client side for formatting. Should I also modify those client side calls?

Reply:
Leave the client-side for the moment. Go with your suggestion of testing for `forge` and falling back to the known locations of `forge` on different platforms. To workaround people getting different results depending on how they started vscode:

https://code.visualstudio.com/docs/terminal/advanced#_process-environment

Reply:
After diving into this for some minutes I think for now it's better to keep the current approach of using the default installation paths for both linux and windows. Trying to determine if `forge` as a global is available introduces too much complexity because you need a strategy for unix as well as windows for doing that. There are libraries (i.e. `which`) that try to achieve that but after a bit of testing it doesnt behave exactly as the `which` unix command. I'd leave it simple for now and see if any users have trouble with it. From foundry docs, it seems that the installation directory is deterministic

Reply:
Okay, lets see how far we get with platform specific known locations.